drivers/infiniband/hw/ntrdma/ntrdma_cmd.c

```diff
@@ -848,7 +848,6 @@ static int ntrdma_cmd_recv_mr_delete(struct ntrdma_dev *dev,
 	ntrdma_rmr_put(rmr);
 
 	wait_for_completion(&done);
-	ntc_flush_dma_channels(dev->ntc);
```

Comment:
Since we poll and not wait for dma completion we need to flush in order not to fail on queue full, you may change to two different modes, and flush only if in poll mode, but not to remove it completely (relevant to all lines removal in this file).